probe-rs/src/architecture/riscv/mod.rs

```diff
@@ -237,10 +237,15 @@ impl<'probe> CoreInterface for Riscv32<'probe> {
         // write 1 to the haltreq register, which is part
         // of the dmcontrol register
 
-        tracing::debug!(
-            "Before requesting halt, the Dmcontrol register value was: {:?}",
-            self.interface.read_dm_register::<Dmcontrol>()?
-        );
+        if tracing::enabled!(tracing::Level::DEBUG) {
```

Comment:
This `enabled!` macro means the `read_dm_register` is skipped if logging to file is disabled.

Reply:
Hmm, I guess we shouldn't need this, but it's a work around for the fact that it's printed regardless of the log level?

It would be really nice to find the root cause, but I think this an acceptable work-around for now.

Reply:
I would consider that a bug in tracing, but fair thing :)